activerecord/lib/active_record/reflection.rb

```diff
@@ -139,6 +139,17 @@ def primary_key_type
         klass.type_for_attribute(klass.primary_key)
       end
 
+      def belongs_to?
+        false
+      end
+
+      def through_reflection
+      end
```

Comment:
The `belongs_to?` and `through_reflection` definition aren't needed here because they're defined in AssociationReflection and all the reflections below (HM, BT, HMT) all inherit from AssociationReflection.